### PR TITLE
feat: add support for static cache config

### PR DESCRIPTION
**Pull request by Ian2012**
### Description

This PR allows the setup of a configurable static cache running on NGINX. Make sure to configure the following snippet to your nginx HTTP config:

```nginx
proxy_cache_path /tmp/nginx_cache levels=1:2 keys_zone=static-cache:32m use_temp_path=off max_size=4g inactive=24h;
proxy_cache_key $scheme$proxy_host$request_uri;
proxy_cache_lock on;
proxy_cache_use_stale updating;
proxy_ignore_headers Cache-Control;
```

### Benchmarks:

Using k6 for fetching multiple static files.

Before:
```
          /\      |‾‾| /‾‾/   /‾‾/   
     /\  /  \     |  |/  /   /  /    
    /  \/    \    |     (   /   ‾‾\  
   /          \   |  |\  \ |  (‾)  | 
  / __________ \  |__| \__\ \_____/ .io

     execution: local
        script: openedx-testing/mfe/request_static_files.js
        output: -

     scenarios: (100.00%) 1 scenario, 400 max VUs, 1m0s max duration (incl. graceful stop):
              * default: 400 looping VUs for 30s (gracefulStop: 30s)

     █ MFE static files

     data_received........: 413 MB 12 MB/s
     group_duration.......: avg=4.46s    min=2.07s    med=4.12s    max=9.52s p(90)=6.37s p(95)=7.88s
     http_req_duration....: avg=2.11s    min=192.17ms med=1.99s    max=6.37s p(90)=3.38s p(95)=3.95s
     http_req_failed......: 0.00%  ✓ 0          ✗ 4704 
     http_req_receiving...: avg=696.22ms min=71.23µs  med=154.75ms max=3.97s p(90)=1.85s p(95)=2.26s
     http_req_waiting.....: avg=1.41s    min=158.02ms med=1.31s    max=4.88s p(90)=2.31s p(95)=2.73s
     http_reqs............: 4704   132.536314/s
     vus..................: 3      min=3        max=400
```
After:
```
          /\      |‾‾| /‾‾/   /‾‾/   
     /\  /  \     |  |/  /   /  /    
    /  \/    \    |     (   /   ‾‾\  
   /          \   |  |\  \ |  (‾)  | 
  / __________ \  |__| \__\ \_____/ .io

     execution: local
        script: openedx-testing/mfe/request_static_files.js
        output: -

     scenarios: (100.00%) 1 scenario, 400 max VUs, 1m0s max duration (incl. graceful stop):
              * default: 400 looping VUs for 30s (gracefulStop: 30s)

     █ MFE static files

     data_received........: 768 MB 22 MB/s
     group_duration.......: avg=1.88s    min=285.36ms med=1.45s    max=12.83s p(90)=4.22s    p(95)=5.14s   
     http_req_duration....: avg=864.04ms min=133.68ms med=333.14ms max=9.8s   p(90)=2.2s     p(95)=3s      
     http_req_failed......: 0.00%  ✓ 0          ✗ 8756 
     http_req_receiving...: avg=617.4ms  min=94.65µs  med=5.2ms    max=8.47s  p(90)=1.92s    p(95)=2.65s   
     http_req_waiting.....: avg=246.54ms min=132.95ms med=154.04ms max=3.15s  p(90)=510.83ms p(95)=768.91ms
     http_reqs............: 8756   253.325541/s
     vus..................: 7      min=7        max=400
```

### CODE REVIEW

File: drydock/plugin.py [hunk 1]
```diff
@@ -162,6 +162,18 @@ def get_sync_waves_for_resource(resource_name: str) -> SYNC_WAVES_ORDER_ATTRS_TY
             "lms-worker",
             "cms-worker",
         ],
+        "NGINX_STATIC_CACHE_CONFIG": {
```
Comment (jfavellar90) [inline drydock/plugin.py:165]: New feature docs are missing

File: drydock/plugin.py [hunk 2]
```diff
@@ -162,6 +162,18 @@ def get_sync_waves_for_resource(resource_name: str) -> SYNC_WAVES_ORDER_ATTRS_TY
...
+            "lms": {
```
Comment (jfavellar90) [inline drydock/plugin.py:166]: We should add this feature to our test suite

File: drydock/templates/drydock/k8s/ingress/static-cache.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: ingress-static-{{ service }}
+  namespace: {{ K8S_NAMESPACE }}
+  annotations:
+    nginx.ingress.kubernetes.io/proxy-body-size: 8m
+    nginx.ingress.kubernetes.io/proxy-buffering: "on"
+    nginx.ingress.kubernetes.io/configuration-snippet: |
+      {{ patch("static-cache-config") | indent(6)}}
```
Comment (jfavellar90) [inline drydock/templates/drydock/k8s/ingress/static-cache.yml:12]: What about adding the content of the file [drydock/patches/static-cache-config](https://github.com/eduNEXT/drydock/pull/106/files#diff-9ba3d1cc82bcdb839c9d60c984b1939023f52f9fb93c8cab0eca43a83f328eb6) here and add an `extra-static-cache-config` patch which is up to the plugin user to implement?

File: drydock/patches/kustomization-resources [hunk 4]
```diff
@@ -12,6 +12,7 @@
 - plugins/drydock/k8s/ingress/cms.yml
 - plugins/drydock/k8s/ingress/mfe.yml
 - plugins/drydock/k8s/ingress/extra-hosts.yml
+{% if DRYDOCK_NGINX_STATIC_CACHE_CONFIG %}- plugins/drydock/k8s/ingress/static-cache.yml{%- endif %}
```
Comment (jfavellar90) [inline drydock/patches/kustomization-resources:15]: In the early stage of this feature, I think it should be disabled by default.

File: README.md [hunk 5]
```diff
@@ -56,6 +56,7 @@ The following configuration options are available:
 - `DRYDOCK_ENABLE_SENTRY`: Whether to enable sentry. Defaults to `true`.
 - `DRYDOCK_SENTRY_DSN`: The sentry DSN. Defaults to `""`.
 - `DRYDOCK_POD_LIFECYCLE`: Whether to enable pod lifecycle. Defaults to `true`.
+- `NGINX_STATIC_CACHE_CONFIG`: A list of dictionaries with settings for different services to cache their assets in NGINX.
```
Comment (jfavellar90) [inline README.md:59]: Could you please include an example of how the NGINX_STATIC_CACHE_CONFIG variable should look like? In this way, the operator knows what keys to add for every service

File: drydock/templates/drydock/k8s/ingress/static-cache.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      proxy_cache_use_stale error timeout updating http_404 http_500 http_502 http_503 http_504;
+      proxy_cache static-cache;
+      proxy_cache_valid any 120m;
+      proxy_cache_bypass $http_x_purge;
+      add_header X-Cache-Status $upstream_cache_status;
+      {{ patch("static-cache-config") | indent(6)}}
+spec:
+  ingressClassName: nginx
+  rules:
+  - host: {{ config["host"] }}
```
Comment (jfavellar90) [inline drydock/templates/drydock/k8s/ingress/static-cache.yml:23]: If we have 2 ingress objects pointing to the same host (let's say `my.host`), but only one ingress object defines the location `/static/`, when visiting https://my.host/static/, has the ingress defining the `/static/` location the priority to process the request?

Comment (Ian2012) [reply drydock/templates/drydock/k8s/ingress/static-cache.yml:23]: Yes, the ingress will end rendered on the nginx pod as:

```nginx
server {
  # Whatever the first ingress setups

  location /static/ {
     # our static settings
  }
}
```

**Review by jfavellar90 — APPROVED** [review]

File: drydock/templates/drydock/k8s/ingress/static-cache.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    nginx.ingress.kubernetes.io/proxy-buffering: "on"
+    nginx.ingress.kubernetes.io/configuration-snippet: |
+      # Cache settings
+      proxy_cache_valid 404 10m;
+      proxy_cache_use_stale error timeout updating http_404 http_500 http_502 http_503 http_504;
+      proxy_cache static-cache;
+      proxy_cache_valid any 120m;
+      proxy_cache_bypass $http_x_purge;
+      add_header X-Cache-Status $upstream_cache_status;
+      {{ patch("static-cache-config") | indent(6)}}
```
Comment (MoisesGSalas) [inline drydock/templates/drydock/k8s/ingress/static-cache.yml:19]: Does nginx play nicely with overrides for this o does it complain if you try to configure the same value twice?

For example you set `proxy_cache_valid any 300m;` in the patch.

Comment (Ian2012) [reply drydock/templates/drydock/k8s/ingress/static-cache.yml:19]: I've tested this, but that should be the expected behavior

Comment (Ian2012) [reply drydock/templates/drydock/k8s/ingress/static-cache.yml:19]: Tested on musashi.chennai. Working as expected

**Pull request merged**